src/redis_stack/general_data_structures.md

```diff
@@ -0,0 +1,320 @@
+```
+
+Notice that the `SADD` command is variadic.
+
+`SREM` is used to remove members of a list. It returns `1` if the member is present in the set, or `0` if it is not.
+
+```redis Remove set members
+SREM bike:1:addons "bell"
+SREM bike:1:addons "sissy bar"
+```
+
+The `SISMEMBER` command is used to test set membership. It returns `1` if the member is present, and `0` otherwise.
+
+```redis Test set membership
+SISMEMBER bike:1:addons "reflectors"
+SISMEMBER bike:1:addons "sissy bar"
+```
+
+Use `SMEMBERS` to return all the members of a set.
+
+```redis Get set members
+SMEMBERS bike:1:addons
+```
+
+`SUNION` combines two or more sets and returns all their elements.
+
+```redis SUNION usage
+SADD bike:1:special_addons "sparkle coat finish" "banana seat"
+SUNION bike:1:addons bike:1:special_addons
+```
+
+Sets have two ways to return one or more random members of a set:
+
+- [SPOP](https://redis.io/commands/spop)
+- [SRANDMEMBER](https://redis.io/commands/srandmember)
+
+Each command takes a key and, optionally, a count as arguments. `SPOP` removes and returns a random member, whereas `SRANDMEMBER` just returns the randomly selected members without removing them.
+
+```redis SPOP/SRANDMEMBER usage
+SMEMBERS bike:1:addons
+SRANDMEMBER bike:1:addons
+SPOP bike:1:addons
+SMEMBERS bike:1:addons
+```
+
+See [here](https://redis.io/commands/?group=set) for the entire list of Redis set-related commands.
+
+## Sorted sets
+
+While handy, Redis sets are unsorted, which limits their usefulness. It is for this reason that Redis sorted sets were added to its stable of data types.
+
+Sorted sets are similar to sets, except each unique member has an associated score.
+
+You can use sorted sets to create, for example, a set of bike brands named after famous computer programmers, where the score element, the programmer's birth year, is used to sort the set.
+
+```redis Create a sorted set
+ZADD bike:brands 1940 "Alan Kay"
+ZADD bike:brands 1906 "Grace Hopper"
+ZADD bike:brands 1953 "Richard Stallman"
+ZADD bike:brands 1965 "Yukihiro Matsumoto"
+ZADD bike:brands 1916 "Claude Shannon"
+ZADD bike:brands 1969 "Linus Torvalds"
+ZADD bike:brands 1957 "Sophie Wilson"
+ZADD bike:brands 1912 "Alan Turing"
+```
+
+Now you can use the [ZRANGE](https://redis.io/commands/zrange) command to retrieve members in the order of birth year.
+
+```redis ZRANGE usage
+ZRANGE bike:brands 2 4
+```
+
+A popular use case for sorted sets is leaderboards for games.
+
+See [here](https://redis.io/commands/?group=sorted-set) for the entire list of Redis sorted set commands.
+
+## Hashes
+
+Lists, sets, and sorted sets cover a lot of ground in terms of use cases, but the [hash](https://redis.io/docs/data-types/hashes) data type is on a higher level. Redis hashes are maps (sequences of key-value pairs) and are used to represent objects in Redis. Consider the following example that shows how to create a hash key using the [HSET](https://redis.io/commands/hset) command.
```

Comment:
Maybe "cover a lot of ground in terms of use cases" can be reworded/streamlined a bit?